### PR TITLE
DAOS-12859 test: use pool and container labels (pass 1)

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: ior DaosAggregationMultiPoolCont BoundaryPoolContainerSpace FullPoolContainerCreate MultipleContainerDelete PerServerFaultDomainTest SnapshotAggregation DaosObjectQuery DaosSnapshotTest ManagementServiceResilience DmvrDstCreate AgentFailure test_basiccheckout_sanity DiskFailureTest,-manual IoSysAdmin ServerRankFailure TargetFailure EcodCellSizeProperty IorCrash PoolAutotestTest TestWithScrubberTargetEviction RbldDeleteObjects OSAOfflineDrain OSAOfflineExtend OSAOfflineParallelTest OSAOfflineReintegration OSAOnlineExtend OSAOnlineReintegration NvmeFaultReintegrate OSAOnlineDrain NvmePoolExtend NvmePoolExclude IoAggregation
Skip-unit-tests: true
Skip-fault-injection-test: true

- Update tests to use pool and container labels instead of uuids.
- Add more TestContainer wrapper functions for daos commands
- Simplify some TestContainer utility logic

Required-githooks: true

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [x] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [x] No new NLT or valgrind warnings.  Check the classic view.
  * [x] Quick-build or Quick-functional is not used.
* [x] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
Failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12049/6/tests:
- deployment/ior_per_rank.py - manual test and not expected to pass in CI- https://daosio.atlassian.net/browse/DAOS-13078
- vmd/fault_reintegration.py - manual test and not expected to pass in CI - https://daosio.atlassian.net/browse/DAOS-13157

File: src/tests/ftest/util/test_utils_container.py [hunk 1]
```diff
@@ -920,6 +907,27 @@ def punch_records(self, indices, punch_dkey=True):
...
+        Returns:
+            str: JSON output of daos container check.
+
+        Raises:
+            DaosTestError: if params are invalid
+            CommandFailure: Raised from the daos command call.
+
+        """
+        if not self.daos:
+            raise DaosTestError("Undefined daos command")
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_container.py:927]: @phender Thoughts on removing these checks now? Nearly all tests have `container.daos` defined now that they use `self.get_container()`. The remaining manual instantiations of `TestContainer` are few and generally API tests that should eventually move to unit tests anyway

**Comment by daltonbohning** [comment]
Failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12049/7/tests
- aggregation/multiple_pool_cont.py
  - Passed in previous run with no changes to this test (force pushed so history lost).
  - Probably unrelated to this PR, but need to check to be safe
- deployment/ior_per_rank.py - manual test and not expected to pass in CI- https://daosio.atlassian.net/browse/DAOS-13078

**Comment by daltonbohning** [comment]
Failures in https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12049/10/tests
- deployment/ior_per_rank.py - manual test and not expected to pass in CI- https://daosio.atlassian.net/browse/DAOS-13078
- vmd/fault_reintegration.py - manual test and not expected to pass in CI - https://daosio.atlassian.net/browse/DAOS-13157

**Review by shimizukko — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Pull request merged**